Sources/CreateAPI/Generator/Templates.swift

```diff
@@ -31,14 +31,14 @@ final class Templates {
     }
 
     func `struct`(name: TypeName, contents: [String], protocols: Protocols) -> String {
-        let lhs = [options.access, "struct", name.rawValue].compactMap { $0 }
+        let lhs = [options.access ?? "", "struct", name.rawValue].filter { !$0.isEmpty }
         let rhs = protocols.sorted()
         return declaration(lhs: lhs, rhs: rhs, contents: contents)
     }
     
     func `class`(name: TypeName, contents: [String], protocols: Protocols) -> String {
         let type = options.entities.isMakingClassesFinal ? "final class" : "class"
-        let lhs = [options.access, type, name.rawValue].compactMap { $0 }
```

Comment:
This is where we combine the `lhs`.

https://github.com/imjn/CreateAPI/blob/b194fc20dff442c3810cc0865038c5543f1ae100/Sources/CreateAPI/Generator/Templates.swift#L46-L54

Reply:
I also had an idea that I should add the `.filter { !$0.isEmpty }` to where we combine them like below ↓

```Swift
private func declaration(lhs: [String], rhs: [String], contents: [String]) -> String {
    let lhs = lhs.joined(separator: " ").filter { $0.isEmpty } ⬅️⬅️⬅️
    let rhs = rhs.joined(separator: ", ")
    return """
    \(rhs.isEmpty ? lhs : "\(lhs): \(rhs)") {
    \(contents.joined(separator: "\n\n").indented)
    }
    """
}
```
With this, we can achieve what we want with less code.
However, I decided to add the filter to `func class()` and `func struct()` as they already filter `nil` and I thought filtering `nil` and empty string should be in the same place and `func class()` should not care about that.

What do you think about this...?

Reply:
Good questions.. More generally, I think maybe we could improve some of this further since there is also the `var access: String` computed property at the top of this file (but we can't use it here since it appends whitespace too).

So for now, I think that what you have done is probably fine 👍

Reply:
I think ideally, `options.access` should never be allowed to be an empty string in the first place either. That's something else we might want to look at 😄